dig.go

```diff
@@ -30,21 +30,36 @@ import (
 
 const (
 	_optionalTag = "optional"
+	_nameTag     = "name"
 )
 
+// Unique identification of an object in the graph.
+type key struct {
+	t    reflect.Type
+	name string
```

Comment:
Yes, added a test.
We can change in the future if for reason we think we should be case insensitive